src/components/common/CountdownTimer.tsx

```diff
@@ -0,0 +1,99 @@
+"use client";
+
+import { useEffect, useMemo, useState } from "react";
+import { target } from "@/data/TargetDate";
+import { google} from "calendar-link";
+import { event } from "@/data/Calendar";
+import React from "react";
+
+interface TimeLeft {
+  days: number;
+  hours: number;
+  minutes: number;
+  seconds: number;
+}
+
+
+export default function CountdownTimer() {
+  const [hackathonTime, setHackathonTime] = useState(false);
+
+
+  const [timeLeft, setTimeLeft] = useState<TimeLeft>({
+    days: 0,
+    hours: 0,
+    minutes: 0,
+    seconds: 0,
+  });
+
+  const timeUnits = [
+    { label: 'Days', value: timeLeft.days },
+    { label: 'Hours', value: timeLeft.hours },
+    { label: 'Minutes', value: timeLeft.minutes },
+    { label: 'Seconds', value: timeLeft.seconds },
+  ];
+
+  
+  const googleUrl = useMemo(() => google(event), [event]); 
+
+
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      const now = new Date();
+      const difference = target.getTime() - now.getTime();
+
+      const d = Math.floor(difference / (1000 * 60 * 60 * 24));
+
+      const h = Math.floor(
+        (difference % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60)
+      );
+
+      const m = Math.floor((difference % (1000 * 60 * 60)) / (1000 * 60));
+
+      const s = Math.floor((difference % (1000 * 60)) / 1000);
+
+      setTimeLeft({ days: d, hours: h, minutes: m, seconds: s });
+
+    }, 1000);
+
+    return () => clearInterval(interval);
+  }, []);
+
+
+  return (
+    <div className="flex flex-col items-center justify-center min-h-screen p-4">
+
+    <h1 className="text-3xl h-0">
+      AUG 2-3, 2025
```

Comment:
Should be linked to the data in the data folder

Reply:
hey Anton I could be really dumb for this but what should I do for this one? Am I just meant to grab the data from the data folder?